plugins/c8y_firmware_plugin/src/main.rs

```diff
@@ -76,7 +77,7 @@ async fn main() -> Result<(), FirmwareManagementError> {
     let fw_plugin_opt = FirmwarePluginOpt::parse();
 
     if fw_plugin_opt.init {
-        init(Path::new(PERSISTENT_DIR_PATH))?;
+        init(Path::new(DEFAULT_DATA_PATH))?;
```

Comment:
You're absolutely right on that "inconsistency" aspect. But it's not as straightforward as passing a `--data-dir` option as this init is usually done during the installation, at a point where the user doesn't really invoke this manually, but rather gets called automatically by the installation script. So, we'll have to provide some default path from that installation script which is most likely gonna be `/var/tedge`. So, the end result will be the same.

So, we can think of this `--init` option as one that creates these directories at the **"default"**  location. We're planning to move away from this static directory creation model where these "necessary directories" will be created as part of the regular execution of the plugin itself rather than relying on them being pre-created as discussed here: https://github.com/thin-edge/thin-edge.io/pull/1776#discussion_r1122176984. So, this concern of yours will be addressed separately in that ticket.

Reply:
> at a point where the user doesn't really invoke this manually,

I have a doubt if it is really true. By debian package installation, yes. However, in docker, user needs to prepare such directories in `Dockerfile`, and I`--init` option is a help for that.

But the latter point, that we will address with #1776 sounds good. In short, we don't need to work on this now, but keep it in mind.

Reply:
I don't think it is worth having the `--data-dir` option as you can actually control the value via the configuration which is already influenced by the `--config-dir` flag. The `config-dir` is a special case because it influences all settings and tedge needs to know where the `tedge.toml` file is before starting things, whereas the `--data-dir` is more a secondary setting relating to more runtime/cache data.

Reply:
To solve this initialisation mess, we'll pass the `--config-dir` option along with the `--init` option when plugins are explicitly initialized so that they can read the `data.dir`,`log.dir` and all other paths from the config directly rather than passing them all as options. But, when `init` is done during the installation, it'll still read the config from the default location (`/etc/tedge`) and end up using the default paths provided there. This is an unpleasent side effect that we will have to deal with until its addressed properly in #1776.